src/LCD_Fonts.h

```diff
@@ -0,0 +1,21 @@
+/**
+ * A library to control the monochrome LCD displays found on Nokia 5110.
+ *
+ * It is a derivative of Jim Lindblom's work, from Sparkfun. The goal
+ * was to decrease its size so it can be ran on ATTiny microcontrollers
+ * that have limited RAM size. Therefore, the main difference is that a
+ * framebuffer, which would consume all dynamic memory, is not used. Get the
+ * original library [here](https://github.com/sparkfun/GraphicLCD_Nokia_5110).
+ *
+ * On how to use this library, please refer to the `examples/` folder.`
+ */
+#pragma once
+#include <Arduino.h>
+
+struct CustomFont {
+    virtual const unsigned char* getFont(char character) = 0;
+    virtual const uint8_t colSize() = 0;
+    virtual const uint8_t rowSize() = 0;
+    virtual const unsigned char* HSpacingChar() = 0;
+    virtual const uint8_t HSpacingSize() = 0;
+};
```

Comment:
Unless I am terribly mistaken, we don't need polymorphism here (and if we did need, `CustomFont` should definitely have a virtual destructor). Do you think the following class can accommodate your use case?

```suggestion
class NokiaLcdFont
{
public:
    NokiaLcdFont(const unsigned char fontTable[][Nokia_LCD_Fonts::kColumns_per_character],
                 const unsigned char* hSpace,
                 uint8_t hSpaceSize,
                 uint8_t characterOffset = 0x20)
        : hSpace{ hSpace }
        , hSpaceSize{ hSpaceSize }
        , columnSize{ Nokia_LCD_Fonts::kColumns_per_character }
        , rowSize{ Nokia_LCD_Fonts::kRows_per_character }
        , kFontTable{ fontTable }
        , kCharacterOffset{ characterOffset }
    {
    }
    const unsigned char* getFont(char character) const
    {
        return *(kFontTable + (character - kCharacterOffset));
    }

    const unsigned char* const hSpace;
    const uint8_t hSpaceSize;
    const uint8_t columnSize;
    const uint8_t rowSize;

private:
    const unsigned char (*kFontTable)[Nokia_LCD_Fonts::kColumns_per_character];
    const uint8_t kCharacterOffset;
};
```

Reply:
It makes perfect sense! It's so valuable to have another point of view. I will makes the changes and test during the week, I let you know. Thanks for the code review :)

Reply:
I get the general idea, using a class to create objects that represent the font is great. But I'm having problems with the font table array. I'm trying to create an LcdFont class, it will have a default constructor that passes Nokia font parameters and another constructor that takes all parameters including a pointer to the font table and that's the problem. Do you have any thoughts on this?

Reply:
Not sure what is the problem you are referring to, but why do you need two constructors?

You can use this class as-is and then create a global constant at `src/Nokia_LCD.cpp` like you have already done. E.g.

```cpp
const LcdFont nokiaFont{kDefault_font, space, 1};
```

And then use it pretty much the way you already do in your pull request.

Reply:
Hi Dimitris, sorry for the delay. I just made it work now. Nevermind the class with two constructors :sweat_smile: I had to do small change in the constructor because the fonts have a different column number and the param that you suggested `const unsigned char fontTable[][Nokia_LCD_Fonts::kColumns_per_character]` won't work.
I also created another method `setDefaultFont()` in order to set the NokiaFont back again ((in case the user is using another font).

Reply:
Hope it is what you expect. I'm open for any improvement anyway.

Reply:
Made a minor change to the `getFont` method of the comment for more readability.